client/src/app/translation-extraction.component.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * Workaround for extraction of localize keys
```

Comment:
work around for extraction of localize keys

Reply:
Maybe we can do something like this?

`<span i18n="@@WARN_ALREADY_MEMBER">User {user} is already a member of this project</span>`

And then manually replacing the substring `{user}` by the username after translation? Haven't tested it yet and it'is even more code to write, but maybe that will work.

Reply:
Angular-CLI  `ng xi18n` genearete different id for `<span i18n="@@WARN_ALREADY_MEMBER">User {user} is already a member of this project</span>`  against `$localize`:@@MESSAGE_ID:User ${ user }:user: is already`

The farmer become such as
```
<source>User <x id="INTERPOLATION" alter="{user}"/> is already a member of this project</source>
```
in where the latter should be
```
<source>User <x id="PH"/> is already a member of this project</source>
```

It cause the record in message catalog become invalid, UI show a broken message such as `@@:MESSAGE_ID:User { George }:user: is already a member of this project` which include id and placeholder name, and logs error which said there is no interpolation placeholder in the text.

You can provide something `fixer` utility to convert placeholder tag which derived from typescript text into correct one, if you want to use it.

A blog describe behavior here: https://blog.ninja-squad.com/2019/12/10/angular-localize/

Reply:
A prospective place holder, INTERPOLATION,  is described here https://angular.io/guide/i18n#translate-plurals-and-alternate-expressions

Reply:
I found a way to hack this, using `$localize:@@MESSAGE_ID:User ${ user }:INTERPOLATION: is already` to force place holder id to "INTERPOLATION"